src/hooks/useDeviceState.ts

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from "react";
+
+const MIN_TABLET_WIDTH = 768;
+const MIN_DESKTOP_WIDTH = 1280;
+
+export enum Device {
+  MOBILE = "mobile",
+  TABLET = "tablet",
+  PC = "pc",
+}
+
+function getDeviceState() {
+  const width = window.innerWidth;
```

Comment:
SSR에서 window가 undefined로 에러가 발생할 상황은 없을까요.

Reply:
SSR에서 getDeviceState가 호출된다면 `window`의 존재를 알지 못해 
`ReferenceError: window is not defined`가 발생합니다.
아래와 같이 수정하였습니다.

```suggestion
  if (typeof window === "undefined") return Device.Mobile;

  const width = window.innerWidth;
```

Reply:
해당 조건문을 여러 곳에서 재사용할 수 있을 것 같은데 별도 상수로 빼면 좋을 것 같아요.